### PR TITLE
Deprecated: Type package

### PR DESCRIPTION
**Pull request by sirreal**
## Description

Add types to the `deprecated` package.

## How has this been tested?
Tests pass.

## Types of changes
New feature: Add type declarations to the `deprecated` package.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/deprecated/README.md [hunk 1]
```diff
@@ -68,7 +68,7 @@ message is only logged once.
 
 _Type_
 
--   `Object` 
+-   `Record<string,(|undefined)>`
```
Comment (sirreal) [inline packages/deprecated/README.md:71]: This wasn't picked up properly.

Comment (gziolo) [reply packages/deprecated/README.md:71]: Interesting, is it logged somewhere?

**Review by gziolo — APPROVED** [review]
Nice 👍

File: packages/deprecated/README.md (hunk 1 above)
Comment (sirreal) [reply packages/deprecated/README.md:71]: It is now 🙂 

https://github.com/WordPress/gutenberg/issues/26971

**Pull request merged**